src/Data/FunctorF.purs

```diff
@@ -0,0 +1,17 @@
+module Data.FunctorF where
+
+import Data.NaturalTransformation (type (~>))
+
+-- | A type class for higher order functors; types of kind `(* -> *) -> * -> *`.
+-- |
+-- | The mapping operation `hoist` allows the inner functor to be changed. This
+-- | is similar to `map` for a `Functor`, only here we are transforming a
+-- | functor rather than a value, so correspondingly we perform this with a
+-- | natural transformation rather than a rank-1 function.
+-- |
+-- | Instances must satisfy the following laws:
+-- |
+-- | - Identity: `hoist id = id`
+-- | - Composition: `hoist (f <<< g) = hoist f <<< hoist g`
+class FunctorF t where
+  hoist :: forall f g. (f ~> g) -> t f ~> t g
```

Comment:
Actually, come to think of it, there is a notion of variance here. `FunctorF Functor` should imply `FunctorF Monad` really, but you can't talk about that. Maybe a class like:

```haskell
class Covariant f where
  covariant :: (c :- d) -> f c :- f d  
```

would be useful. But now I am definitely getting off topic.